draft-irtf-cfrg-opaque.md

```diff
@@ -976,16 +976,25 @@ handshake_encrypt_key =
 
 Nh is the output length of the underlying hash function.
 
-The HKDF input parameter `info` is computed as follows:
+The Derive-Secret parameter `info` is computed as:
 
 ~~~
-info = "3DH keys" || I2OSP(len(client_nonce), 2) || client_nonce
-                  || I2OSP(len(server_nonce), 2) || server_nonce
-                  || I2OSP(len(client_identity), 2) || client_identity
-                  || I2OSP(len(server_identity), 2) || server_identity
+info = "3DH" || client_preamble || client_transcript ||
+                server_preamble || server_transcript)
 ~~~
 
-See {{identities}} for more information about identities client_identity and server_identity.
+where `client_preamble` and `server_preamble` are computed as:
+
+~~~
+client_preamble = I2OSP(len(client_identity), 2) || client_identity
+server_preamble = I2OSP(len(server_identity), 2) || server_identity
+~~~
+
+`client_transcript` KE1 and `server_transcript` is KE2 excluding KE2.enc_server_info
```

Comment:
client_transcript **is**  KE1

Reply:
👍 good catch

Reply:
The new `client_transcript` variable only appears twice. Can we just rename it as `KE1` in the two occurrences it appears in, rather than introducing a new variable?

Also, same thing with the `server_transcript` variable. I am wary of introducing all of these new terms that aren't accompanied by a struct definition. Perhaps we could abstract away the struct definition of `KE2` to denote an "inner" struct which captures these values that we want to represent as `server_transcript` here? So in other words, something like:

```
struct {
  CredentialResponse response;
  uint8 server_nonce[32];
  uint8 server_keyshare[Npk];
} InnerKE2;

struct {
  InnerKE2 inner_ke2;
  opaque enc_server_info<0..2^16-1>;
  uint8 mac[Nh];
} KE2;
```

And then just use `InnerKe2` here instead of `server_transcript`.